crates/indexer/src/models/token_datas.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use crate::{
+    schema::token_datas,
+    util::{hash_str, u64_to_bigdecimal},
+};
+use anyhow::Context;
+use aptos_api_types::WriteTableItem as APIWriteTableItem;
+use bigdecimal::BigDecimal;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Queryable, Serialize)]
+#[primary_key(creator_address, collection_name_hash, name_hash, transaction_version)]
+#[diesel(table_name = "token_datas")]
+pub struct TokenData {
+    pub creator_address: String,
+    pub collection_name_hash: String,
+    pub name_hash: String,
+    pub collection_name: String,
+    pub name: String,
+    pub transaction_version: i64,
+    pub maximum: bigdecimal::BigDecimal,
```

Comment:
I had made the special types to enforce the limits afair

Reply:
Nope. It was this I looked.